### PR TITLE
libcxx v13.0.1 (redux)

### PR DESCRIPTION
**Pull request by h-vetinari**
Rebases #80 against master

Closes #82 
Closes #80 
Closes #79

### CODE REVIEW

**Comment by h-vetinari** [comment]
Waiting for builds from https://github.com/conda-forge/clangdev-feedstock/pull/163

**Comment by h-vetinari** [comment]
Unfortunately, the [comment ](https://github.com/conda-forge/libcxx-feedstock/pull/80#issuecomment-979034614) from the PR for 13.0.0 is still relevant - it keeps failing while trying to find `#include <__config_site>`, even when setting `LLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF` as mentioned in https://reviews.llvm.org/D97572.

The upstream build system was reshuffled a fair bit, so it's possible that what this feedstock is doing is currently broken, and _maybe_ considered unsupported by upstream. Certainly, the [docs](https://libcxx.llvm.org/BuildingLibcxx.html) for building libcxx look very different than what we're doing, so I'm a bit at a loss on how to fix it.

I can raise an issue upstream, but first wanted to hear if someone from @conda-forge/libcxx has a better idea what's going on.

PS. While trying to fix the failure, I switched the include paths from `include/c++/v1` (the current [default](https://libcxx.llvm.org/BuildingLibcxx.html#cmdoption-arg-libcxx-install-include-dir-path)) to `include` (at least, I _think_ we want unversioned include paths...), see https://github.com/conda-forge/libcxx-feedstock/pull/83/commits/eadb6edf161d83c8cecbfca1bd39dd9cb371d55d

**Comment by h-vetinari** [comment]
OK, the problem seems to be a different one - the headers are getting installed in the wrong directory:
```
Install the project...
-- Install configuration: "Release"
-- Installing: $SRC_DIR/build/include/__algorithm/adjacent_find.h
-- Installing: $SRC_DIR/build/include/__algorithm/all_of.h
-- Installing: $SRC_DIR/build/include/__algorithm/any_of.h
[...]
-- Installing: $SRC_DIR/build/include/__config_site
-- Installing: $PREFIX/lib/libc++.so.1.0
-- Installing: $PREFIX/lib/libc++.so.1
-- Installing: $PREFIX/lib/libc++.so
-- Installing: $PREFIX/lib/libc++.a
-- Installing: $PREFIX/lib/libc++experimental.a
```

That's despite `-DCMAKE_INSTALL_PREFIX` being set as [documented](https://libcxx.llvm.org/BuildingLibcxx.html)...

**Comment by h-vetinari** [comment]
OK, that SRC_DIR stuff was my derp. But I just noticed the following [commit](https://github.com/llvm/llvm-project/commit/6f17768e11480063f4c2bcbeea559505fee3ea19), which says:
> The [libcxxabi] atandalone build has been deprecated since LLVM 14, and it is not supported anymore. Please use one of the ways described at https://libcxx.llvm.org/BuildingLibcxx.html for building libc++.

The [docs](https://releases.llvm.org/13.0.1/projects/libcxx/docs/BuildingLibcxx.html) for 13.x say:
```
$ git clone https://github.com/llvm/llvm-project.git
$ cd llvm-project
$ mkdir build
$ cmake -G Ninja -S llvm -B build -DLLVM_ENABLE_PROJECTS="libcxx;libcxxabi" # Configure
$ ninja -C build cxx cxxabi                                                 # Build
$ ninja -C build check-cxx check-cxxabi                                     # Test
$ ninja -C build install-cxx install-cxxabi                                 # Install
```
(which is in line with the mainline recommendation, except that nowadays there's also `libunwind`)

I'm not sure why we have the build setup we currently have, and whether it'd be reasonable to change over to that (would need full llvm sources, rather than just libcxx).

**Comment by h-vetinari** [comment]
OK, chasing down `__config_site` a bit more, it get's generated [here](https://github.com/llvm/llvm-project/blob/llvmorg-13.0.1/libcxx/include/CMakeLists.txt#L391), relative to `LIBCXX_GENERATED_INCLUDE_TARGET_DIR`, which gets set [here](https://github.com/llvm/llvm-project/blob/llvmorg-13.0.1/libcxx/CMakeLists.txt#L441-L451) (for `LLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF`).

This made me finally understand why the recipe was failing... Fixed now. 🥳

**Comment by h-vetinari** [comment]
@conda-forge/libcxx 
This is now _finally_ ready after having one crucial lightbulb go on.

**Comment by h-vetinari** [comment]
@conda-forge/libcxx 
I think this should be the last missing piece before we can [activate](https://github.com/conda-forge/clang-compiler-activation-feedstock/pull/77) llvm 13 in conda-forge. Could you PTAL?

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -8,10 +8,10 @@ c_compiler:
   - vs2017           # [win]
 vc:
   - 14
-c_compiler_version:     # [osx]
-  - "11.1.0"            # [osx]
-cxx_compiler_version:   # [osx]
-  - "11.1.0"            # [osx]
+c_compiler_version:     # [linux]
```
Comment (xhochy) [inline recipe/conda_build_config.yaml:11]: Can you add a comment here why this is needed?

Comment (h-vetinari) [reply recipe/conda_build_config.yaml:11]: It's in the commit message of https://github.com/conda-forge/libcxx-feedstock/pull/83/commits/ae977380098cf25245b1b5752a2384aa00821e7f, but sure.

**Comment by h-vetinari** [comment]
Gentle ping @conda-forge/libcxx

**Review by xhochy — APPROVED** [review]
Looks good but I'm not a maintainer here, so someone else should press the button.

**Comment by h-vetinari** [comment]
Another very gentle ping @conda-forge/libcxx (I'd be joining as maintainer - then I could also stop pinging for routine tasks)

**Comment by isuruf** [comment]
Thanks

**Pull request merged**